packages/combobox/src/useCombobox.ts

```diff
@@ -0,0 +1,777 @@
+    inputValue: _inputValue,
+    getToggleButtonProps: getDownshiftTriggerProps,
+    getInputProps: getDownshiftInputProps,
+    getMenuProps: getDownshiftListboxProps,
+    getItemProps: getDownshiftOptionProps,
+    closeMenu: closeListbox,
+    openMenu: openListbox,
+    setHighlightedIndex: setActiveIndex,
+    selectItem: setDownshiftSelection
+  } = useDownshift<OptionValue | OptionValue[]>({
+    id: prefix,
+    toggleButtonId: `${prefix}-trigger`,
+    menuId: `${prefix}-listbox`,
+    getItemId: index => `${prefix}-option-${index}`,
+    items: values,
+    inputValue,
+    initialInputValue,
+    itemToString: transformValue as any /* HACK around Downshift's generic type overuse */,
+    selectedItem: selectionValue,
+    initialSelectedItem: initialSelectionValue,
+    isOpen: isExpanded,
+    defaultIsOpen: defaultExpanded,
+    initialIsOpen: initialExpanded,
+    highlightedIndex: activeIndex,
+    defaultHighlightedIndex: _defaultActiveIndex,
+    initialHighlightedIndex: initialActiveIndex,
+    onStateChange: handleDownshiftStateChange,
+    stateReducer,
+    environment: doc.defaultView || window
+  });
+
+  const {
+    getLabelProps: getFieldLabelProps,
+    getHintProps,
+    getInputProps: getFieldInputProps,
+    getMessageProps
+  } = useField({ idPrefix, hasHint, hasMessage });
+
+  /*
+   * Effects
+   */
+
+  useLayoutEffect(
+    () => {
+      // Trigger autocomplete selection override. Use layout effect to prevent
+      // `defaultActiveIndex` flash.
+      if (isAutocomplete && _isExpanded && _selectionValue && !matchValue) {
+        const value = Array.isArray(_selectionValue)
+          ? _selectionValue[
+              _selectionValue.length - 1 // multiselectable most recent
+            ]
+          : _selectionValue;
+        const index = values.findIndex(current => current === value);
+
+        if (index !== -1) {
+          setActiveIndex(index);
+        } else if (_defaultActiveIndex !== undefined) {
+          setActiveIndex(_defaultActiveIndex);
+        }
+      }
+    },
+    /* eslint-disable-line react-hooks/exhaustive-deps */ [
+      /* matchValue, // prevent match active index reset */
+      isAutocomplete,
+      _isExpanded,
+      _selectionValue,
+      _inputValue,
+      values,
+      _defaultActiveIndex,
+      setActiveIndex
+    ]
+  );
+
+  useEffect(
+    // Downshift does not expect a dropdown like Garden's combobox where the
+    // wrapping div functions like an open/close button. Finesse state so that
+    // default close-on-blur functionality is not undone by the toggle of the
+    // trigger.
+    () => setTriggerContainsInput(triggerRef.current?.contains(inputRef.current)),
+    [triggerRef, inputRef]
+  );
+
+  useEffect(() => {
+    // Clear the select-only match value after the the user stops typing for
+    // half a second.
+    clearTimeout(matchTimeoutRef.current);
+    matchTimeoutRef.current = window.setTimeout(() => setMatchValue(''), 500);
```

Comment:
Actually there is a TypeScript difference, which is why `window` is used here (consistent with similar code throughout Garden). Since this is effect code, we don't need to be concerned with SSR implications.